api-server/src/venues/dto/create-update-venue-request.ts

```diff
@@ -49,6 +51,12 @@ export class CreateVenueRequest {
   @ApiProperty({ example: 'Downtown' })
   @IsOptional()
   neighborhood?: string;
+
+  constructor(copy?: Partial<CreateVenueRequest>) {
+    if (isNotNullOrUndefined(copy)) {
+      cloneAttributes<CreateVenueRequest>(copy, this);
+    }
+  }
```

Comment:
Not following why this is necessary

Reply:
I didn't want to mutate state on the values passed into fillInGPSCoordinatesIfNeededWhenPossible. Unfortunately because NestJs makes you use classes rather than `types` you can't simply shallow clone an object with the spread operator.

This is something I've done in several places on the backend to basically give me a away to clone an object, while maintaining it's class.

Honestly I hate it 🤣 but I haven't come up with a better way

Reply:
I wonder if some of the newly introduced clone methods like https://developer.mozilla.org/en-US/docs/Web/API/structuredClone, might help, haven't had time to dig back into this for a bit 🤷

Reply:
I had assumed there was a built-in way to do this.